platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/LdapService.java

```diff
@@ -0,0 +1,103 @@
+package ddf.ldap.ldaplogin;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.karaf.jaas.config.JaasRealm;
+import org.apache.karaf.jaas.config.impl.Config;
+import org.apache.karaf.jaas.config.impl.Module;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LdapService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LdapService.class);
+
+    private static final String CONFIG_NAME = "ldap";
+
+    private Config config;
+
+    private BundleContext context;
+
+    private List<Module> modules;
+
+    public LdapService() {
+        config = new Config();
+        modules = new ArrayList<>();
+        context = getContext();
+        config.setBundleContext(context);
+        config.setName(CONFIG_NAME);
+        config.setRank(2);
+        config.setModules(new Module[] {});
+
+        LOGGER.debug("Registering new service as a JaasRealm.");
+        context.registerService(JaasRealm.class, config, null);
+    }
+
+    /**
+     * Updates an existing ldap module with a new one or adds a
+     * new module to the list of existing modules.
+     *
+     * @param newModule that will replace a module or be added to the list of modules.
+     */
+    public synchronized void update(Module newModule) {
+        boolean isExistingModule = false;
+        if (modules.isEmpty()) {
+            modules.add(newModule);
+        } else {
+            for (int i = 0; i < modules.size(); i++) {
+                if (modules.get(i)
+                        .getName()
+                        .equals(newModule.getName())) {
+                    modules.set(i, newModule);
+                    isExistingModule = true;
+                    break;
+                }
+            }
+            if (!isExistingModule) {
+                modules.add(newModule);
+            }
+        }
+        config.setModules(modules.toArray(new Module[modules.size()]));
+    }
+
+    /**
+     * Delete an ldap module given its id.
+     *
+     * @param id of the module.
+     * @return true, if the delete was successful, false otherwise.
+     */
+    public synchronized boolean delete(String id) {
+        boolean isModuleDeleted = false;
+        for (int i = 0; i < modules.size(); i++) {
+            if (modules.get(i)
+                    .getName()
+                    .equals(id)) {
+                modules.remove(i);
+                isModuleDeleted = true;
+                break;
+            }
+        }
+        return isModuleDeleted;
+    }
+
+    /**
+     * Return the list of created ldap modules.
+     *
+     * @return list of modules.
+     */
+    public List<Module> getModules() {
+        return this.modules;
```

Comment:
I'm not in love with leaking this list out and not making a `Collections.unmodifiableList()` out of it. You're synchronizing the two methods that modify it but this exposes it to callers, such as they are.